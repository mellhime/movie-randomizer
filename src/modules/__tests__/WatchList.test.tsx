import { render, renderHook } from "@testing-library/react";

import { WatchList } from "@modules";
import { useFetch } from "@hooks";

describe("WatchList component", () => {
  it("should match snapshot", () => {
    const { asFragment } = render(<WatchList />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("useFetch hook", () => {
  it("should get data from api", () => {
    const { result } = renderHook(() => useFetch("watchList"));

    expect(result.current[0][0]).toHaveProperty("title");
  });
});
