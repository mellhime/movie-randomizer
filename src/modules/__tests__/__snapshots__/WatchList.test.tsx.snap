// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WatchList component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="p-datatable p-component p-datatable-responsive-scroll"
    data-pc-name="datatable"
    data-pc-section="root"
    data-scrollselectors=".p-datatable-wrapper"
    data-showgridlines="false"
    pr_id_1=""
  >
    <div
      class="p-datatable-wrapper"
      data-pc-section="wrapper"
      style="overflow: auto;"
    >
      <table
        class="p-datatable-table"
        data-pc-section="table"
        role="table"
        style="min-width: 50rem;"
      >
        <thead
          class="p-datatable-thead"
          data-pc-name="datatable"
          data-pc-section="thead"
          role="rowgroup"
        >
          <tr
            data-pc-section="headerrow"
            role="row"
          >
            <th
              data-p-filter-column="false"
              data-p-frozen-column="false"
              data-p-reorderable-column="false"
              data-p-resizable-column="false"
              data-p-sortable-column="false"
              data-pc-name="datatable"
              data-pc-section="headercell"
              role="columnheader"
            >
              <div
                class="p-column-header-content"
                data-pc-section="headercontent"
              >
                <span
                  class="p-column-title"
                  data-pc-section="headertitle"
                >
                  Title
                </span>
              </div>
            </th>
            <th
              data-p-filter-column="false"
              data-p-frozen-column="false"
              data-p-reorderable-column="false"
              data-p-resizable-column="false"
              data-p-sortable-column="false"
              data-pc-name="datatable"
              data-pc-section="headercell"
              role="columnheader"
            >
              <div
                class="p-column-header-content"
                data-pc-section="headercontent"
              >
                <span
                  class="p-column-title"
                  data-pc-section="headertitle"
                >
                  Edit button
                </span>
              </div>
            </th>
          </tr>
        </thead>
        <tbody
          class="p-datatable-tbody"
          data-pc-section="tbody"
          role=" rowgroup"
        >
          <tr
            class="p-datatable-emptymessage"
            data-pc-section="emptymessage"
            role="row"
          >
            <td
              colspan="2"
              data-pc-name="datatable"
              data-pc-section="bodycell"
              role="cell"
            >
              No available options
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;
